contracts/Pool.sol

```diff
@@ -711,11 +839,60 @@ contract Erc20Pool is Pool {
     /// @param amount The amount to top up with
     function topUp(uint128 amount) public {
         if (amount == 0) return;
-        erc20.transferFrom(msg.sender, address(this), amount);
-        onTopUp(amount);
+        transferToContract(amount);
+        topUpSuspending(amount);
+    }
+
+    /// @notice Updates all the sender parameters of the sender of the message.
+    ///
+    /// Tops up and withdraws unsent funds from the balance of the sender.
+    /// The sender must first grant the contract a sufficient allowance to top up.
+    /// Sends the withdrawn funds to the sender of the message.
+    ///
+    /// Sets the target amount sent on every block from the sender of the message.
+    /// On every block this amount is rounded down to the closest multiple of the sum of the weights
+    /// of the receivers and proxies and split between them proportionally to their weights.
+    /// Each receiver and proxy then receives their part from the sender's balance.
+    /// If set to zero, stops funding.
+    ///
+    /// Sets the weight of the provided receivers and proxies of the sender of the message.
+    /// The weight regulates the share of the amount sent on every block
+    /// that each of the sender's receivers and proxies get.
+    /// Setting a non-zero weight for a new receiver or
+    /// a new proxy adds it to the list of the sender's receivers.
+    /// Setting zero as the weight for a receiver or a proxy
+    /// removes it from the list of the sender's receivers.
+    /// @param topUpAmt The topped up amount
+    /// @param withdraw The amount to be withdrawn, must not be higher than available funds.
+    /// Can be `WITHDRAW_ALL` to withdraw everything.
+    /// @param amountPerBlock The target amount to be sent on every block.
+    /// Can be `AMOUNT_PER_BLOCK_UNCHANGED` to keep the amount unchanged.
+    /// @param updatedReceivers The list of the updated receivers and their new weights
+    /// @param updatedProxies The list of the updated proxies and their new weights
```

Comment:
I’m totally fine referencing the documentation of `updateSenderInternal` here. This avoids duplicating the documentation and it getting out of sync.

Reply:
I'm not sure if it's worth deduplication. There are some issues with it:

- `updateSenderInternal` is `internal`, will its doc strings be easily accessible to the consumers?
- `updateSenderInternal` can't say that it sends any funds, it returns the amount which should be sent
- Different `updateSender`s handle incoming funds slightly differently which leads to different lists of parameters (message amount) and different prerequisites (allowance)

I'm afraid that we'll turn 3 self-containing texts into a web of references, amendments and conditions.

Reply:
I don’t meant that we *just* reference `updateSenderInternal`. We should definitely include the differences. But for the parameter documentation for example and the rest that is shared we can just refer to `updateSenderInternal`.

> `updateSenderInternal` is `internal`, will its doc strings be easily accessible to the consumers?

At the moment we don’t produce any output and the documentation is only consumed by reading the comments.

This is just a suggestion though. If you prefer, we can keep it as it is.

Reply:
I'd prefer leaving it as it is. I to don't like the redundancy, but forcing users to mental stitching of the function docs is IMO even worse.